File: output/llama3.1-8b/avatar/Python/Java/atcoder_ABC171_E.java
import java.util.*;
import java.util.function.*;

public class atcoder_ABC171_E {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = Integer.parseInt(scanner.nextLine());
        String[] input = scanner.nextLine().split(" ");
        int[] a = new int[N];
        for (int i = 0; i < N; i++) {
            a[i] = Integer.parseInt(input[i]);
        }
        int total = Arrays.stream(a).reduce(0, (x, y) -> x ^ y);
        StringBuilder sb = new StringBuilder();
        for (int ai : a) {
            sb.append((ai ^ total)).append(" ");
        }
        System.out.println(sb.toString().trim());
    }
}

File: output/llama3.1-8b/avatar/Java/Python/atcoder_ABC171_E.py
import sys
from typing import List

def main():
    n = int(input())
    a = [int(x) for x in input().split()]
    r = 0
    for i in range(n):
        r ^= a[i]
    print(*[r ^ ai for ai in a])

if __name__ == "__main__":
    main()